client/src/legacy/GridField.js

```diff
@@ -120,8 +120,16 @@ $.entwine('ss', function($) {
         }
       }, ajaxOpts));
     },
-    showDetailView: function(url) {
-      window.location.href = url;
+    showDetailView: function(url, event) {
+      this.maybeOpenUrlInNewTab(event, url, () => window.location.href = url);
+    },
+    maybeOpenUrlInNewTab: function(event, url, openInSameTabCallback) {
```

Comment:
I'd just name it `openUrl`, the method name doesn't always need to convey all the logic of the method body in its name.